dyno-jedis/src/main/java/com/netflix/dyno/jedis/DynoJedisClient.java

```diff
@@ -3255,8 +3269,74 @@ public Builder withDiscoveryClient(DiscoveryClient client) {
             return this;
         }
 
+        public Builder withShadowClusterName(String shadowCluster) {
+            shadowClusterName = shadowCluster;
+            return this;
+        }
+
+        public Builder withShadowHostSupplier(HostSupplier shadowHostSupplier) {
```

Comment:
Good question! We actually do need two HostSupplier objects b/c there is a 1:1 relationship with a HostSupplier and the application they are retrieving host information for. Note however that there is only a single DiscoveryClient object that gets shared between both HostSupplier objects.